src/tokens/AssetList.tsx

```diff
@@ -100,6 +101,8 @@ export default function AssetList({
     sortedTokensWithBalanceOrShowZeroBalanceSelector(state, supportedNetworkIds)
   )
 
+  const hideWalletBalances = useSelector(hideWalletBalancesSelector)
```

Comment:
```suggestion
  const hideWalletBalances = useSelector(hideWalletBalancesSelector) && isWalletTab
```
we could do this to be extra safe, but in practice should never make a difference as long as the old assets screen cannot be accessed when the tab navigator is on.